tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -207,6 +210,10 @@ def _fetch_resources(self, query):
 
         return resources
 
+    @classmethod
+    def _handle_child_retrieval_exception(cls, child_retrieval_func):
```

Comment:
this feels a bit hacky/messy. a cleaner path would be to customize the source class which does the actual query for this specific exception handling and in the resource subclass and hand back via get_source override. ie use the existing interface instead of introducing another one.

Reply:
Done, thanks